simphony/testing/abc_check_engine.py

```diff
@@ -177,15 +177,17 @@ def test_delete_dataset(self):
             name = "test_" + str(i)
             engine.add_dataset(self.create_dataset(name=name))
 
+        datasets_and_names = [(ds, ds.name) for ds in engine.iter_datasets()]
+
         # delete each of the datasets
-        for ds in engine.iter_datasets():
-            engine.remove_dataset(ds.name)
+        for (dataset, name) in datasets_and_names:
+            engine.remove_dataset(name)
             # test that we can't get deleted datasets
             with self.assertRaises(ValueError):
-                engine.get_dataset(ds.name)
+                engine.get_dataset(name)
             # test that we can't use the deleted datasets
             with self.assertRaises(Exception):
-                self.compare_dataset(ds, ds)
+                self.compare_dataset(dataset, dataset)
```

Comment:
I don't think this error is related to JYULB wrapper specifically. Have you tried this test for any other wrapper? I think it will always fail. If I'm mistaken, then sure we can make a new issue for it and try to solve it there.